### DISCUSSION TITLE
Support other data types than int and float

### DISCUSSION THREAD

**Discussion by arpa42** [citrusvanilla/tinyflux#52] [Ideas]
I'm starting to use tinyflux and I noticed in the documentation that the field values can only be `int` or `float`.

I find this very restrictive. What are the reasons for this restriction?

Do you think that adding `bool` or `str` is possible ?

**Comment by citrusvanilla** [comment]
You should use Tags for those types.

**Reply by arpa42** [reply]
tags are not always the most appropriate field. Imagine a sensor that returns a true/false value, the value is true/false, and it's not a tag. This value does not need to be indexed.

Also, the documentation states that tinyflux follows the principles of InfluxDB, which accepts these data types as values (c.f. [influxdb glossary](https://docs.influxdata.com/influxdb/v1/concepts/glossary/#field-value))

If necessary, I can participate or work on a PR. 
I remain open to the possible reasons why these types of data are not supported.

**Reply by citrusvanilla** [reply]
I'd be happy to review a PR for this work.  But you'll have to code to the spec in the link:

```Field values are not indexed - queries on field values scan all points that match the specified time range and, as a result, are not performant.```
